objectherkenning_openbare_ruimte/on_edge/detection_pipeline/components/model_result.py

```diff
@@ -0,0 +1,314 @@
+        blur_kernel_size : int (default: 165)
+            Kernel size (used for both width and height) for GaussianBlur.
+        box_padding : int (default: 0)
+            Optional: increase box by this amount of pixels before applying the blur.
+
+        Return
+        ------
+        image : numpy.ndarray
+            The image blurred.
+        """
+        img_height, img_width, _ = self.image.shape
+
+        for box in boxes:
+            x_min, y_min, x_max, y_max = map(int, box)
+
+            x_min = max(0, x_min - box_padding)
+            y_min = max(0, y_min - box_padding)
+            x_max = min(img_width, x_max + box_padding)
+            y_max = min(img_height, y_max + box_padding)
+
+            logger.debug(f"Blurring inside: {(x_min, y_min)} -> {(x_max, y_max)}")
+            area_to_blur = self.image[y_min:y_max, x_min:x_max]
+            blurred = cv2.GaussianBlur(
+                area_to_blur, (blur_kernel_size, blur_kernel_size), 0
+            )
+            self.image[y_min:y_max, x_min:x_max] = blurred
+
+    def blur_outside_boxes(
+        self,
+        boxes: Union[List[Tuple[float, float, float, float]], npt.NDArray[np.float_]],
+        blur_kernel_size: int = 165,
+        box_padding: int = 0,
+    ):
+        """
+        Apply GaussianBlur with given kernel size to the area outside the given bounding box(es).
+
+        Parameters
+        ----------
+        image : numpy.ndarray
+            The image to blur.
+        boxes : List[Tuple[float, float, float, float]]
+            Bounding box(es) outside which to blur, in the format (xmin, ymin, xmax, ymax).
+        blur_kernel_size : int (default: 165)
+            Kernel size (used for both width and height) for GaussianBlur.
+        box_padding : int (default: 0)
+            Optional: increase box by this amount of pixels before applying the blur.
+        """
+        img_height, img_width, _ = self.image.shape
+
+        blurred_image = cv2.GaussianBlur(
+            self.image, (blur_kernel_size, blur_kernel_size), 0
+        )
+
+        for box in boxes:
+            x_min, y_min, x_max, y_max = map(int, box)
+
+            x_min = max(0, x_min - box_padding)
+            y_min = max(0, y_min - box_padding)
+            x_max = min(img_width, x_max + box_padding)
+            y_max = min(img_height, y_max + box_padding)
+
+            logger.debug(f"Blurring outside: {(x_min, y_min)} -> {(x_max, y_max)}")
+            blurred_image[y_min:y_max, x_min:x_max] = self.image[
+                y_min:y_max, x_min:x_max
+            ]
+
+        self.image = blurred_image
+
+    def crop_outside_boxes(
+        self,
+        boxes: Union[List[Tuple[float, float, float, float]], npt.NDArray[np.float_]],
+        box_padding: int = 0,
+        fill_bg: bool = False,
+    ) -> List[npt.NDArray[np.int_]]:
+        """
+        Crop image to the area(s) given by the yolo annotation box(es).
+
```

Comment:
i don't understand the role of this function

Reply:
We don't use it yet, but one of the scenarios for data minimisation is to crop the target objects only. This function does that.